### PR TITLE
Setup Travis CI and various badges

### PR DESCRIPTION
**Pull request by tkuhlman**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - go get -u github.com/golang/dep/...
+  - dep ensure
+  - go get github.com/mattn/goveralls
+
+script:
+  - go test -v -race ./...
+  - go test -coverprofile=coverage.txt ./...
+
+after_success:
+  - goveralls -coverprofile=coverage.txt -service=travis-ci
```
Comment (scniro) [inline .travis.yml]: 💯

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+language: go
+go:
+  - 1.10.2
+
+services:
+  - docker
+sudo: required
```
Comment (scniro) [inline .travis.yml]: have you tried `false` here? I know when true travis spins up a whole VM vs. a container if false. I know it's needed for some stuff so I'd be curious what the need is

Comment (scniro) [reply .travis.yml]: Looks like we can go containers, https://docs.travis-ci.com/user/reference/overview/#Virtualization-environments

Comment (tkuhlman) [reply .travis.yml]: good point I don't know if it is needed at all or not. I'll try without.

Comment (scniro) [reply .travis.yml]: Sweet, `sudo: false` should be the only change here and you'll notice a huge increase in build time

**Pull request merged**